Tchap/Managers/UniversalLink/UniversalLinkService.swift

```diff
@@ -0,0 +1,184 @@
+                                                                case .success:
+                                                                    NSLog("[UniversalLinkService] handleUserActivity. Email successfully validated.")
+                                                                    
+                                                                    if let nextLink = params.queryParams?[Constants.emailValidationNextLinkKey] {
+                                                                        // Continue the registration with the passed nextLink
+                                                                        NSLog("[UniversalLinkService] handleUserActivity. Complete registration with nextLink")
+                                                                        let nextLinkURL = URL(string: nextLink)
+                                                                        if let fragment = nextLinkURL?.fragment {
+                                                                            _ = self.handleFragment(fragment, completion: completion)
+                                                                        }
+                                                                    } else {
+                                                                        // No nextLink means validation for binding a new email
+                                                                        NSLog("[UniversalLinkService] handleUserActivity. TODO: Complete email binding")
+                                                                    }
+                                                                case .failure(let error):
+                                                                    NSLog("[UniversalLinkService] handleUserActivity. Error: submitToken failed")
+                                                                    completion(MXResponse.failure(error))
+                                                                }
+                        })
+                    case .failure(let error):
+                        completion(MXResponse.failure(error))
+                    }
+                }
+            }
+            return true
+        } else if let fragment = url.fragment {
+            return self.handleFragment(fragment, completion: completion)
+        }
+        return false
+    }
+    
+    func handleFragment(_ fragment: String, completion: @escaping (MXResponse<UniversalLinkServiceParsingResult>) -> Void) -> Bool {
+        NSLog("[UniversalLinkService] handleFragment: \(fragment)")
+        
+        // Extract required parameters from the link
+        let params = parseFragment(fragment)
+        if params.pathParams.isEmpty {
+            NSLog("[UniversalLinkService] handleFragment: Error: No path parameters")
+            return false
+        }
+        
+        var isSupported: Bool = false
```

Comment:
Prefer use `let isSupported: Bool` to be sure to cover every case

Reply:
done